bleak/backends/bluezdbus/manager.py

```diff
@@ -668,6 +670,11 @@ async def _wait_for_services_discovery(self, device_path: str) -> None:
 
         If a disconnect happens before the completion a BleakError exception is raised.
         """
+        if device_path not in self._properties:
+            raise BleakError(
+                f"Device '{device_path}' disappeared while waiting to discover services"
+            )
```

Comment:
If we don't check here, we could end up with an unretrieved exception when calling `_wait_condition` since it will raise `KeyError`